keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/CanaryConstraintDeployHandler.kt

```diff
@@ -27,6 +28,7 @@ interface CanaryConstraintDeployHandler {
    */
   suspend fun deployCanary(
     constraint: CanaryConstraint,
+    artifact: DeliveryArtifact,
     version: String,
     deliveryConfig: DeliveryConfig,
     targetEnvironment: Environment,
```

Comment:
had to add `artifact` to this method signature as in the EC2 case we'll need the base OS. It seems kinda weird that the artifact wasn't already being passed.

I considered making this class have a type parameter for the specific sub-type of `DeliveryArtifact` but it was a bit of a PITA.